### ISSUE TITLE
Block "File" not accessible in frontend

### ISSUE THREAD

**Issue by 00travelgirl00** [WordPress/gutenberg#22908]
**Describe the bug**
Download Button in File Block is not accessible for screen reader users. 

The Download Block shows the link of the file and a button to download. Both are not connected with each other, also the download attribute is missing. 

Screen reader users will only hear the "Link [name of the link]" and "Link [Download]". So they are missing out the information which file they are gonna to download. 

**To reproduce**

1. Create a File block
2. Use a screenreader and go over the block in the frontend

**Expected behavior**
you should here "download [name of file]"

**Screenshots**

<img width="984" alt="Bildschirmfoto 2020-06-04 um 16 48 16" src="https://user-images.githubusercontent.com/21318377/83786101-8da41c00-a692-11ea-9f7e-987e60c02120.png">
<img width="1095" alt="Bildschirmfoto 2020-06-04 um 16 48 30" src="https://user-images.githubusercontent.com/21318377/83786113-91d03980-a692-11ea-81cb-2c307275ec0a.png">

**Desktop (please complete the following information):**
 - OS and Voice Over 

WordPress 5.4.1 
Gutenberg Plugin 8.2.1

**Possible Solution

add an ID to the link of the file and add a Aria Description to the Button with the ID of the file

**Comment by gziolo** [comment]
@enriquesanchez or @afercia, it sounds like something that should be addressed.

If my assumption is confirmed, it will require deprecation entry in the block definition. We won’t be able to apply such fix back to already published posts but it’s be auto-fixed when saving edited posts.

**Comment by enriquesanchez** [comment]
Thanks @gziolo!

@jeryj [already started a PR](https://github.com/WordPress/gutenberg/pull/22914
) and I believe he shared some of the same concerns you mentioned.
